lib/walk-sync/index.js

```diff
@@ -1,18 +1,21 @@
 var fs = require('graceful-fs')
 var path = require('path')
+var assign = require('../util/assign')
 
-var walkSync = function (dir, filelist) {
-  var files = fs.readdirSync(dir)
-  filelist = filelist || []
+var walkSync = function (dir, list) {
+  var files = fs.readdirSync(path.resolve(dir))
+  list = list || []
   files.forEach(function (file) {
     var nestedPath = path.join(dir, file)
-    if (fs.lstatSync(nestedPath).isDirectory()) {
-      filelist = walkSync(nestedPath, filelist)
+    var stat = fs.lstatSync(nestedPath)
+    if (stat.isDirectory()) {
+      list.push({path: nestedPath, stats: assign({}, {type: 'dir'}, stat)})
+      list = walkSync(nestedPath, list)
     } else {
-      filelist.push(nestedPath)
+      list.push({path: nestedPath, stats: assign({}, {type: 'file'}, stat)})
```

Comment:
Oh I forgot to tell you that I added that just to make it more distinct. If you think that's unnecessary, I remove it right away.